### ISSUE TITLE
[bug] 02-Calling-an-API yarn start doesn't work on fresh checkout

### ISSUE THREAD

**Issue by blakedietz** [auth0-samples/auth0-react-samples#133]
On a fresh checkout after installing dependencies `yarn start` fails with the following error.

```bash
 yarn start
yarn run v1.16.0
$ npm-run-all --parallel spa server
$ react-scripts start
$ node server.js

There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "babel-jest": "24.7.1"

Don't try to install it manually: your package manager does it automatically.
However, a different version of babel-jest was detected higher up in the tree:

  /Users/blakedietz/projects/auth0-react-samples/02-Calling-an-API/node_modules/babel-jest (version: 24.8.0)

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "babel-jest" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if /Users/blakedietz/projects/auth0-react-samples/02-Calling-an-API/node_modules/babel-jest is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls babel-jest in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed babel-jest.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.

P.S. We know this message is long but please read the steps above :-) We hope you find them helpful!

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "spa" exited with 1.
error Command failed with exit code 1.
```

As instructed running the following

```bash
rm -rf node_modules yarn.lock;
yarn;
yarn start;
```

Works. It's jus a bummer that you have to do this on a fresh checkin.

**Comment by stevehobbsdev** [comment]
@blakedietz Thanks for reporting this, looking now. Can you confirm if it's just sample 02 that does this, or is sample 01 affected also?

**Comment by stevehobbsdev** [comment]
Turns out this is part of a wider issue to do with how `react-scripts@3.0.0` pinned the version of `babel-jest`. They've patched this in `3.0.1`; I've raised #135 to fix this issue.

Thanks for the issue report!

**Comment by blakedietz** [comment]
This was for both. Thanks for looking into this.
